edgedb/retry.py

```diff
@@ -10,16 +10,25 @@ def __init__(self, retry, owner, iteration):
         super().__init__(owner, retry._options.transaction_options)
         self.__retry = retry
         self.__iteration = iteration
+        self.__started = False
 
     async def start(self):
         if not self._managed:
             raise errors.InterfaceError(
                 "Only managed retriable transactions are supported. "
                 "Use `async with transaction:`"
             )
-        await self._start(single_connect=self.__iteration != 0)
+
+    async def _ensure_transaction(self):
+        await self.start()
```

Comment:
Hm, `start()` has no awaits in it now... And does pretty much nothing too.

Reply:
Right, it's a bit weird here - I was just trying to avoid overriding `__aenter__` here. Let me extract the `raise InterfaceError` part in `start()` to a new sync method.

Reply:
We can keep `start()` to explicitly open a transaction. I think it's OK. (We've chatted about this in Slack)

Reply:
Please elaborate why an explicit `start()` is necessary.

Reply:
That comment was before we opened the discussion about this api. I think that we should strip all manual control menthols from the retrying transaction.